index.d.ts

```diff
@@ -0,0 +1,226 @@
+/// <reference types="node" />
+import { Readable, Writable, Transform } from 'stream';
+import { EventEmitter } from 'events';
+import { Agent as HttpAgent } from 'http';
+import { Agent as HttpsAgent } from 'https';
+
+import AbortController from 'abort-controller'
+import type * as fetch from 'node-fetch';
+
+declare function megajs(options: megajs.StorageOpts, cb?: megajs.errorCb): megajs.Storage;
+
+declare namespace megajs {
+
+    export function megaEncrypt(key: Buffer, options?: cryptOpts): void | Transform;
+    export function megaDecrypt(key: Buffer, options?: cryptOpts): Transform;
+    export function megaVerify(key: Buffer): void | Transform;
+
+    export class Storage extends EventEmitter {
+        api: API;
+        key: Buffer;
+        sid: string;
+        aes: AES;
+        name: string;
+        user: any; // Not sure
+        email: string;
+        shareKeys: { [nodeId in string]: Buffer };
+        options: StorageOpts;
+        status: StorageStatus;
+        root: MutableFile;
+        trash: MutableFile;
+        inbox: MutableFile;
+        mounts: ReadonlyArray<File>[];
+        files: { [id in string]: MutableFile };
+        RSAPrivateKey: (number | number[])[]; // tsc generated this        
+        constructor(options: StorageOpts, cb?: errorCb);
+        toJSON(): StorageJSON;
+        close(cb: noop): void;
+        static fromJSON(json: StorageJSON): Storage;
+        mkdir(opt: mkdirOpts | string, cb?: errorCb): void;
+        upload(opt: uploadOpts | string, buffer?: BufferString, cb?: uploadCb): Writable;
+        login(cb: (error: err, storage: this) => void): void;
+        getAccountInfo(cb: (error: err, account: accountInfo) => void): accountInfo;
+        // "A required parameter cannot follow an optional parameter."
+        // Do check this because in source code force is the first argument but I had to change the order because of above error
+        reload(cb: (error: err, mount: ReadonlyArray<File>[], force?: boolean) => void): void | this;
+        on(event: 'add', listener: (File: MutableFile) => void): this;
+        on(event: 'move', listener: (file: MutableFile, oldDir: MutableFile) => void): this;
+        on(event: 'ready', listener: (storage: this) => void): this;
+        on(event: 'update', listener: (file: MutableFile) => void): this;
+        on(event: 'delete', listener: (file: Readonly<File>) => void): this;
```

Comment:
yes but if the its `MutableFile` then people could do something like `file.rename()` which doesn't make sense because the file has been deleted so it can't ben renamed?